### PR TITLE
Reduced some C++ compiler warnings

### PR DESCRIPTION
**Pull request by sgavil**
Some warnings removed #6189

### CODE REVIEW

**Comment by hcho3** [comment]
Syncing this pull request with the latest `master` branch, to fix the test farm.

**Comment by hcho3** [comment]
Thank you for your contribution, and congrats for your first pull request here! I will review once all unit tests complete.

**Review by trivialfis — APPROVED** [review]
Thanks for the contribution!  Please rebase to latest master branch.

File: src/tree/updater_quantile_hist.h [hunk 1]
```diff
@@ -442,7 +442,7 @@ class QuantileHistMaker: public TreeUpdater {
     // list of nodes whose histograms would be built explicitly.
     std::vector<ExpandEntry> nodes_for_explicit_hist_build_;
 
-    enum DataLayout { kDenseDataZeroBased, kDenseDataOneBased, kSparseData };
+    enum class DataLayout { kDenseDataZeroBased, kDenseDataOneBased, kSparseData };
```
Comment (hcho3) [inline src/tree/updater_quantile_hist.h:445]: This line caused compilation to fail. Why was `DataLayout` changed to an enum class? Does it address a compiler warning?

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: src/tree/updater_quantile_hist.h (hunk 1 above)
Comment (sgavil) [reply src/tree/updater_quantile_hist.h:445]: Yes, it was producing an enum type unscoped warning. I changed the lines where I saw it was used like this: 

https://github.com/dmlc/xgboost/pull/6197/files/b8b00f57582b9e98e28a0506d58011810cbe1762#diff-10aac201acca00858b390ff0d679434fL1239-R1239

Comment (hcho3) [reply src/tree/updater_quantile_hist.h:445]: Got it. Can you fix the failing build for `testxgboost` (C++ tests with Google Test framework)? You can build it by following instructions in https://xgboost.readthedocs.io/en/latest/contrib/unit_tests.html#running-gtest.

Comment (hcho3) [reply src/tree/updater_quantile_hist.h:445]: @sgavil How it is going? Do you need some help?

Comment (sgavil) [reply src/tree/updater_quantile_hist.h:445]: @hcho3 Well, I had to do some refactoring in the names of the unit tests (tbh not sure if I'm able to do that) from example from `ASSERT_EQ(this->data_layout_, RealImpl::kSparseData);` to `ASSERT_EQ(this->data_layout_, RealImpl::DataLayout::kSparseData);` but that belongs to [dmlc-core](https://github.com/dmlc/dmlc-core) and I'm not sure how to proceed with that.

File: tests/cpp/c_api/test_c_api.cc [hunk 2]
```diff
@@ -128,15 +128,15 @@ TEST(CAPI, JsonModelIO) {
   learner->UpdateOneIter(0, p_dmat);
   BoosterHandle handle = learner.get();
 
-  std::string modelfile_0 = tempdir.path + "/model_0.json";
+  std::string modelfile_0 = tempdir.path_ + "/model_0.json";
```
Comment (hcho3) [inline tests/cpp/c_api/test_c_api.cc:131]: Let's not rename the member fields here, as it's outside the scope of this pull request (reduce compiler warning).

**Review by hcho3 — CHANGES_REQUESTED** [review]

File: src/tree/updater_quantile_hist.cc [hunk 3]
```diff
@@ -1138,9 +1138,9 @@ void QuantileHistMaker::Builder<GradientSumT>::FindSplitConditions(
     // split_cond = -1 indicates that split_pt is less than all known cut points
     CHECK_LT(upper_bound,
              static_cast<uint32_t>(std::numeric_limits<int32_t>::max()));
-    for (uint32_t i = lower_bound; i < upper_bound; ++i) {
-      if (split_pt == gmat.cut.Values()[i]) {
-        split_cond = static_cast<int32_t>(i);
+    for (uint32_t bound = lower_bound; bound < upper_bound; ++bound) {
+      if (split_pt == gmat.cut.Values()[bound]) {
+        split_cond = static_cast<int32_t>(bound);
```
Comment (hcho3) [inline src/tree/updater_quantile_hist.cc:1143]: Why are we renaming the loop variable here? The name `i` is commonly used as a loop variable, and there's no need to rename it here. I also don't think the renaming fits the purpose of this pull request (reduce compiler warning).

**Comment by hcho3** [comment]
@sgavil To my knowledge, a C++ compiler does not throw a warning about naming convention. Please remove all variable renaming, unless it's necessary for removing a compiler warning.

File: src/tree/updater_quantile_hist.cc (hunk 3 above)
Comment (sgavil) [reply src/tree/updater_quantile_hist.cc:1143]: It does cause a warning because it is inside another foor loop which has declared another `i`as iterator

Comment (hcho3) [reply src/tree/updater_quantile_hist.cc:1143]: Got it. We can keep this one then.

**Review by hcho3 — APPROVED** [review]

**Pull request merged**